### PR TITLE
Added obsolete documentation and migration steps for IManageMessageFailures

### PR DESCRIPTION
**Pull request by indualagarsamy**
Relates to https://github.com/Particular/docs.particular.net/issues/1281

In version 5, in order to override the behavior of SLR and moving messages to error queue, the user can extend the IManageMessageFailures interface and do custom logic there. See v5 code. 

**Version 5 code**
```c#
public class CustomFaultManager : IManageMessageFailures, INeedInitialization
{
    public void SerializationFailedForMessage(TransportMessage message, Exception e)
    {
        Console.WriteLine("CustomFaultManager - SerializationFailedForMessage");
    }

    public void ProcessingAlwaysFailsForMessage(TransportMessage message, Exception e)
    {
        Console.WriteLine("CustomFaultManager - ProcessingAlwaysFailsForMessage");
    }

    public void Init(Address address)
    {
        Console.WriteLine("CustomFaultManager - Init");
    }
        
    public void Customize(BusConfiguration configuration)
    {
        configuration.RegisterComponents(c => c.ConfigureComponent<CustomFaultManager>(DependencyLifecycle.InstancePerCall));
    }
}
```

In Version 6, this is done by adding a new message processing behavior in the pipeline. 
User can control if the behavior should still invoke SLR or not, by turning off SLR or customizing it a the endpoint level. 
User can control if the message can be forwarded to the error queue or not in the behavior. 

This PR replaces the work @petersgiles and I have been doing on  https://github.com/Particular/docs.particular.net/pull/1300

--
Out of scope for this PR:

@andreasohlund - We might need another PR that you wanted to address customizing error handling in general.: What you mentioned before was:
- To get notified when message is retried or sent to error - Handled by the notifications. [We have doco for this already](http://docs.particular.net/nservicebus/errors/subscribing-to-error-notifications).
- Control the number of FLR's per message type/ex - Can this be done with a custom SLR policy?
- Control the number and delay of SLR's per message type/ex - Can this be done with a custom SLR policy?
- Send a given message to a different "error queue" - Add a behavior to the logical receive pipeline that try/catch and bus.Sends to the give queue (We'll need to make changes to our core API to address this)

### CODE REVIEW

File: nservicebus/pipeline/customizing-error-handling.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#### Rolling Back 
+
+If it's required to rollback the receive operation instead of either handling the message or to forward it to the error queue, invoke `AbortReceiveOperation` as shown below: 
+
+snippet: RollbackMessage
+
+
+### Registering the Behavior
+
+In the example below, the behavior `CustomErrorHandlingBehavior` is being registered in the message handling pipeline. This behavior will be invoked after NServiceBus has invoked the first-level retries.
```
Comment (mikeminutillo) [inline nservicebus/pipeline/customizing-error-handling.md]: > This behavior will be invoked after NServiceBus has invoked the first-level retries.

This is a little bit misleading. It invokes your behavior before the FLR/SLR bits but you don't have any code running before the call to `next()` so all of your logic happens after.  I can't think of a more accurate way to state what you are going for yet.

File: nservicebus/pipeline/customizing-error-handling.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+snippet: RollbackMessage
+
+
+### Registering the Behavior
+
+In the example below, the behavior `CustomErrorHandlingBehavior` is being registered in the message handling pipeline. This behavior will be invoked after NServiceBus has invoked the first-level retries. 
+
+snippet: RegisterCustomErrorHandlingBehavior
+            
+## Versions 5 and Below
```
Comment (mikeminutillo) [inline nservicebus/pipeline/customizing-error-handling.md]: If this is a migration article, does this content belong here? This make it seem less like a migration doco and more like full "customizing error handling" doco (which I'm fine with)

**Comment by mikeminutillo** [comment]
This is really good :+1:

File: Snippets/Snippets_5/Features/CustomFaultManager.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    #region CustomFaultManager
+    public class CustomFaultManager : IManageMessageFailures
+    {
+        public void SerializationFailedForMessage(TransportMessage message, Exception e)
+        {
+            // implement any custom steps for this message when the failure was due to deserialization
+        }
+
+        public void ProcessingAlwaysFailsForMessage(TransportMessage message, Exception e)
+        {
```
Comment (weralabaj) [inline Snippets/Snippets_5/Features/CustomFaultManager.cs]: Is it called before first SLR? Once?

File: nservicebus/pipeline/customizing-error-handling.md (hunk 1 above)
Comment (timbussmann) [inline nservicebus/pipeline/customizing-error-handling.md]: Maybe something like:
The following snippet shows how to register a custom behavior to only receive exceptions which aren't already handled by the retry mechanisms. By placing it after the "MoveFaultsToErrorQueue" step, NServiceBus will continue to send exceptions not handled by the custom behavior to the error queue.

or 

The following snippet shows how to register a custom behavior to ensure that built-in retry mechanism try to handle exceptions first.

Comment (indualagarsamy) [inline nservicebus/pipeline/customizing-error-handling.md]: @timbussmann - I like your second suggestion. I'll go with that.

File: Snippets/Snippets_5/Features/CustomFaultManager.cs (hunk 3 above)
Comment (indualagarsamy) [inline Snippets/Snippets_5/Features/CustomFaultManager.cs]: @weralabaj - For V5, SLR won't be invoked. When you implement this extension, you're taking over the fault managing process. I've clarified this better now.

Comment (indualagarsamy) [inline Snippets/Snippets_5/Features/CustomFaultManager.cs]: If you're interested in a working sample, you can check out:https://github.com/indualagarsamy/CustomFaultManagerMigration

**Comment by indualagarsamy** [comment]
@Particular/docs-maintainers - I've addressed @mikeminutillo and @weralabaj's questions. Review and merge.

**Pull request merged**